app/controllers/api/blueprints_controller.rb

```diff
@@ -33,44 +28,10 @@ def publish_resource(type, id, data)
 
     private
 
-    def create_bundle(blueprint, bundle)
-      blueprint.create_bundle(deserialize_bundle(bundle))
-    rescue => e
-      raise BadRequestError, "Couldn't create the bundle - #{e}"
-    end
-
-    def update_bundle(blueprint, bundle)
-      blueprint.update_bundle(deserialize_bundle(bundle))
-    rescue => e
-      raise BadRequestError, "Couldn't update the bundle - #{e}"
-    end
-
-    def deserialize_bundle(bundle)
-      options = {}
-      if bundle.key?("service_catalog")
-        options[:service_catalog] = service_catalog_options(bundle)
-      end
-      if bundle.key?("service_dialog")
-        options[:service_dialog] = service_dialog_options(bundle)
-      end
-      options[:service_templates] = bundle.fetch("service_templates", []).collect do |st|
-        resource_search(parse_id(st, :service_templates), :service_templates, ServiceTemplate)
-      end
-      options[:entry_points] = bundle["automate_entrypoints"] if bundle["automate_entrypoints"]
-      options
-    end
-
-    def service_catalog_options(bundle)
-      if bundle["service_catalog"]
-        resource_search(parse_id(bundle["service_catalog"], :service_catalogs),
-                        :service_catalogs, ServiceTemplateCatalog)
-      end
-    end
-
-    def service_dialog_options(bundle)
-      if bundle["service_dialog"]
-        resource_search(parse_id(bundle["service_dialog"], :service_dialogs), :service_dialogs, Dialog)
-      end
+    def validate_ui_properties(ui_properties)
+      raise BadRequestError unless ui_properties.key?('service_catalog') &&
+                                   ui_properties.key?('service_dialog') &&
+                                   ui_properties.key?('automate_entrypoints') && ui_properties.key?('ChartDataModel')
```

Comment:
Agreeing with @imtayadeway, maybe chart_data_model.  Also, all BadRequestError should have some kind of error message with them, otherwise it's unclear to the caller as to what could be wrong.